src/component/contents/__tests__/DraftEditorBlock.react-test.js

```diff
@@ -106,401 +116,385 @@ function getProps(block, decorator) {
-        .slice(0, boldLength)
-        .map(c => CharacterMetadata.applyStyle(c, 'BOLD'))
-        .concat(characters.slice(boldLength));
-
-      helloBlock = helloBlock.set('characterList', characters.toList());
-
-      var props = getProps(helloBlock);
-      var block = ReactTestUtils.renderIntoDocument(
-        <DraftEditorBlock {...props} />,
-      );
-
-      var rendered = reactComponentExpect(block)
-        .expectRenderedChild()
-        .toBeComponentOfType('div');
-
-      assertLeaves(rendered, [
-        {
-          text: 'he',
-          offsetKey: 'a-0-0',
-          start: 0,
-          styleSet: BOLD,
-          isLast: false,
-        },
-        {
-          text: 'llo',
-          offsetKey: 'a-0-1',
-          start: 2,
-          styleSet: NONE,
-          isLast: true,
-        },
-      ]);
-    });
-  });
+test('must render multiple leaf nodes', () => {
+  const boldLength = 2;
+  let helloBlock = getHelloBlock();
+  let characters = helloBlock.getCharacterList();
+  characters = characters
+    .slice(0, boldLength)
+    .map(c => CharacterMetadata.applyStyle(c, 'BOLD'))
+    .concat(characters.slice(boldLength));
+
+  helloBlock = helloBlock.set('characterList', characters.toList());
+
+  const props = getProps(helloBlock);
+  const block = ReactTestUtils.renderIntoDocument(
+    <DraftEditorBlock {...props} />,
+  );
+
+  const rendered = reactComponentExpect(block)
+    .expectRenderedChild()
+    .toBeComponentOfType('div');
+
+  assertLeaves(rendered, [
+    {
+      text: 'he',
+      offsetKey: 'a-0-0',
+      start: 0,
+      styleSet: BOLD,
+      isLast: false,
+    },
+    {
+      text: 'llo',
+      offsetKey: 'a-0-1',
+      start: 2,
+      styleSet: NONE,
+      isLast: true,
+    },
+  ]);
+
+  expect(ReactDOM.findDOMNode(block)).toMatchSnapshot();
+});
 
-  describe('Allow/reject component updates', () => {
-    it('must allow update when `block` has changed', () => {
-      var helloBlock = getHelloBlock();
-      var props = getProps(helloBlock);
+test('must allow update when `block` has changed', () => {
+  const helloBlock = getHelloBlock();
+  const props = getProps(helloBlock);
 
-      var container = document.createElement('div');
-      ReactDOM.render(<DraftEditorBlock {...props} />, container);
+  const container = document.createElement('div');
+  ReactDOM.render(<DraftEditorBlock {...props} />, container);
 
-      expect(mockLeafRender.mock.calls.length).toBe(1);
+  expect(mockLeafRender.mock.calls.length).toMatchSnapshot();
 
-      var updatedHelloBlock = helloBlock.set('text', 'hxllo');
-      var nextProps = getProps(updatedHelloBlock);
+  const updatedHelloBlock = helloBlock.set('text', 'hxllo');
+  const nextProps = getProps(updatedHelloBlock);
 
-      expect(updatedHelloBlock).not.toBe(helloBlock);
-      expect(props.block).not.toBe(nextProps.block);
+  expect(updatedHelloBlock !== helloBlock).toMatchSnapshot();
+  expect(props.block !== nextProps.block).toMatchSnapshot();
```

Comment:
The rationale behind was to avoid having tests which would be mixing "snapshots" and normal "assertions" from jest, in case they are mixed up that could become error prone for reviewing. 

Had to force boolean beacuse of:
* snapshot would not be able to capture reference comparison
* Avoiding having tests that have both snapshots and normal assertions in their body.

*** 

Happy to convert that back to a normal assertion if you think that would be better suited

Reply:
That's a good reason - thanks for explaining :) Let's try snapshots all the way, unless we have some test which can't easily be covered with a snapshot for some reason.

Reply:
There is one case on this PR I did not convert to snapshot it was when having assertions for `.throws` throwing errors (Maybe there is a way to snapshot that i will have to think more about it, can follow up in the future with other prs)